editors/code/src/scopes_mapper.ts

```diff
@@ -0,0 +1,79 @@
+import * as vscode from 'vscode';
+import { TextMateRuleSettings } from './scopes';
+
+let mappings = new Map<string, string[]>();
+
+const defaultMapping = new Map<string, string[]>([
+    [
+        'comment',
+        [
+            'comment',
+            'comment.block',
+            'comment.line',
+            'comment.block.documentation'
+        ]
+    ],
+    ['string', ['string']],
+    ['keyword', ['keyword']],
+    ['keyword.control', ['keyword.control', 'keyword', 'keyword.other']],
+    [
+        'keyword.unsafe',
+        ['storage.modifier', 'keyword.other', 'keyword.control', 'keyword']
+    ],
+    ['function', ['entity.name.function']],
+    ['parameter', ['variable.parameter']],
+    ['constant', ['constant', 'variable']],
+    ['type', ['entity.name.type']],
+    ['builtin', ['variable.language', 'support.type', 'support.type']],
+    ['text', ['string', 'string.quoted', 'string.regexp']],
+    ['attribute', ['keyword']],
+    ['literal', ['string', 'string.quoted', 'string.regexp']],
+    ['macro', ['support.other']],
```

Comment:
`support.other` is weird. I believe `['entity.name.function', 'keyword.other', 'entity.name.macro']` would be closer in the spirit of TM?

Reply:
@lnicola You're right, I'll change it!

Reply:
Fixed in https://github.com/rust-analyzer/rust-analyzer/pull/2061/commits/25537d294cb7a3e01d2329a7d07b469d734fc829